src/middleware/router.js

```diff
@@ -321,31 +316,35 @@ function sendRequestToRoutes (ctx, routes, next) {
 
     Promise.all(promises).then(() => {
       logger.info(`All routes completed for transaction: ${ctx.transactionId}`)
-      ctx.state.requestPromise.then(() => {
-        ctx.state.responsePromise.then(() => {
+      messageStore.initiateResponse(ctx, () => {
+        messageStore.completeResponse(ctx, () => {}).then(() => {
```

Comment:
The original code here was meant to ensure that the initiateResponse and competeResponse methods in MessageStore.js only run after the initiateRequest method, which is invoked in another module (streamingRouter.js). 
This is important because the document is created by initiateRequest, and updated by completeRequest, initiateResponse and completeResponse (and so must already exist when these methods execute).
I'm not seeing how that condition is fulfilled here. Is it not necessary anymore?

Reply:
Oh yeah, that functionality is not necessary anymore. It resulted in some methods being called so many times, we would end up updating the same transaction with the same info so many times